packages/react/src/components/LocaleModal/LocaleModal.js

```diff
@@ -41,9 +43,16 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
 
   useEffect(() => {
     (async () => {
-      const locale = await LocaleAPI.getLocale();
-      const list = locale && (await LocaleAPI.getList(locale));
-      const getLangDisplay = await LocaleAPI.getLangDisplay();
+      let list, getLangDisplay;
+      if (localeData && localeDisplay) {
+        list = Object.assign({}, localeData);
+        getLangDisplay = localeDisplay;
+      } else {
+        const locale = await LocaleAPI.getLocale();
+        list = locale && (await LocaleAPI.getList(locale));
+        getLangDisplay = await LocaleAPI.getLangDisplay();
```

Comment:
Weighing in; Just wondered if avoiding critical path with sequential promise refs would be desirable (if not my apologies):

```suggestion
        const pair = await Promise.all([LocaleAPI.getLocale(), LocaleAPI.getLangDisplay()]);
        const locale = pair[0];
        list = locale && (await LocaleAPI.getList(locale));
        getLangDisplay = pair[1];
```

Reply:
@asudoh thank you for the suggestion! Just added it.